packages/grafana-ui/src/components/uPlot/Plot.tsx

```diff
@@ -117,12 +141,150 @@ export class UPlotChart extends React.Component<PlotProps, UPlotChartState> {
+    const dx = dt * PIXELS_PER_MS;
+    let horValue = 0;
+    let vertValue = 0;
+
+    if (this.keyboardState.pressedKeys.ArrowUp) {
+      vertValue -= dx;
+    }
+    if (this.keyboardState.pressedKeys.ArrowDown) {
+      vertValue += dx;
+    }
+    if (this.keyboardState.pressedKeys.ArrowLeft) {
+      horValue -= dx;
+    }
+    if (this.keyboardState.pressedKeys.ArrowRight) {
+      horValue += dx;
+    }
+    if (this.keyboardState.pressedKeys.Shift) {
+      horValue *= SHIFT_MULTIPLIER;
+      vertValue *= SHIFT_MULTIPLIER;
+    }
+
+    this.moveCursor(horValue, vertValue);
+
+    const cursor = this.state.plot.cursor;
+    if (this.keyboardState.pressedKeys[' '] && cursor) {
+      const drawHeight = Number(this.state.plot.over.style.height.slice(0, -2));
+
+      this.state.plot.setSelect(
+        {
+          left: cursor.left! < this.keyboardState.dragStartX! ? cursor.left! : this.keyboardState.dragStartX!,
+          top: 0,
+          width: Math.abs(cursor.left! - (this.keyboardState.dragStartX ?? cursor.left!)),
+          height: drawHeight,
+        },
+        false
+      );
+    }
+
+    this.keyboardState.keysLastHandledAt = time;
+    window.requestAnimationFrame(this.handlePressedKeys);
+  };
+
+  moveCursor = (dx: number, dy: number) => {
+    if (this.state.plot?.cursor.left === undefined) {
+      return;
+    }
+
+    const { cursor } = this.state.plot;
+    this.state.plot.setCursor({
+      left: cursor.left! + dx,
+      top: cursor.top! + dy,
+    });
+  };
+
+  handleFocus = () => {
+    // Is there a more idiomatic way to do this?
+    const drawWidth = Number(this.state.plot?.over.style.width.slice(0, -2));
+    const drawHeight = Number(this.state.plot?.over.style.height.slice(0, -2));
+
+    this.state.plot?.setCursor({ left: drawWidth / 2, top: drawHeight / 2 });
+  };
+
+  handleBlur = () => {
+    this.keyboardState = initialKeyboardState;
+    this.state.plot?.setSelect({ left: 0, top: 0, width: 0, height: 0 }, false);
+  };
+
   render() {
     return (
       <div style={{ position: 'relative' }}>
-        <div ref={this.plotContainer} data-testid="uplot-main-div" />
+        <ThemeContext.Consumer>
+          {(theme) => {
+            const styles = getStyles(theme);
+            return (
+              <div
+                className={styles.focusStyle}
+                ref={this.plotContainer}
+                data-testid="uplot-main-div"
+                tabIndex={0}
+                onFocusCapture={this.handleFocus}
+                onBlur={this.handleBlur}
+                onKeyDown={this.handleKeys}
+                onKeyUp={this.handleKeyRelease}
+              />
+            );
+          }}
+        </ThemeContext.Consumer>
         {this.props.children}
       </div>
     );
   }
 }
+
+const getStyles = stylesFactory((theme: GrafanaTheme2) => ({
+  focusStyle: css({
+    borderRadius: theme.shape.borderRadius(1),
+    '&:focus-visible': getFocusStyles(theme),
+  }),
+}));
```

Comment:
Not sure about this, the focus should be on the panel, not the inner chart. 
when we get started on this issue (using new panel chrome & menu), the full panel will get focused (and show a new menu when it's focused), so tabbing should move focus to menu, then next panel (ideally), with this it would move to inner chart before to next panel. But not sure how we can capture the key down events without this inner component having focus. But maybe having two panel focusable elements is ok for now (outer panel, then menu, then inner visualization) 
https://github.com/grafana/grafana/issues/31080